native/screens/AccountSettings.jsx

```diff
@@ -0,0 +1,272 @@
+        }
+      }, 2000);
+    } else {
+      setConfirmedPassword(null);
+    }
+  }, [newPasswordForm]);
+
+  useEffect(() => {
+    if (!newPasswordsMatch && !confirmedPassword) {
+      setActiveUpdateButton(false);
+    } else {
+      setActiveUpdateButton(true);
+    }
+  }, []);
+
+  const handlePasswordUpdate = async () => {
+    const res = await updatePassword(newPasswordForm.newPassword, userID); // i feel that backend should also error check to confirm current password
+    if (res) {
+      setUpdateStatus('Success');
+      dispatch(uiActions.updateUser(res.user));
+      handleToggle(toggleRerender);
+    } else {
+      setUpdateStatus('Failed');
+    }
+  };
+
+  let UpdatePasswordStatusModal;
+  switch (updateStatus) {
+    case 'Success':
+      UpdatePasswordStatusModal = (
+        <Modal>
+          <View>
+            <Text>Password updated!</Text>
+          </View>
+        </Modal>
+      );
+      break;
+    case 'Failed':
+      UpdatePasswordStatusModal = (
+        <Modal>
+          <View>
+            <Text>Incorrect Password, try again</Text>
+            <Button title="OK" onPress={() => setUpdateStatus('Pending')} />
+          </View>
+        </Modal>
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  const ConfirmedPasswordMessage = ({ status }) => {
+    if (status) {
+      return <Text>Password is confirmed</Text>;
+    } else return <Text>INCORRECT PASSWORD</Text>;
+  };
+
+  // this feels WET
+  return (
+    <View>
+      {updateStatus !== 'Pending' && UpdatePasswordStatusModal}
+      <Text>Confirm your password:</Text>
+      <TextInput
+        onChangeText={(currPassword) =>
+          handleTextChange(currPassword, 'currentPassword', setNewPasswordForm)
+        }
+        value={newPasswordForm.currentPassword}
+      />
+      {confirmedPassword !== null && <ConfirmedPasswordMessage status={confirmedPassword} />}
+      <Text>Create new password:</Text>
+      <TextInput
+        onChangeText={(newPassword) =>
+          handleTextChange(newPassword, 'newPassword', setNewPasswordForm)
+        }
+        value={newPasswordForm.newPassword}
+      />
+      <Text>Confirm new password:</Text>
+      <TextInput
+        onChangeText={(confirmNewPassword) =>
+          handleTextChange(confirmNewPassword, 'confirmNewPassword', setNewPasswordForm)
+        }
+        value={newPasswordForm.confirmNewPassword}
+      />
+
+      {newPasswordForm.newPassword !== newPasswordForm.confirmNewPassword && (
+        <Text>NEW PASSWORD AND CONFIRM NEW PASSWORD DO NOT MATCH</Text>
+      )}
+
+      <Button
+        title="Update my password"
+        disabled={!activeUpdateButton}
+        onPress={handlePasswordUpdate}
+      />
+    </View>
+  );
+};
+
+const DeleteModal = ({ userID, email, navigation, toggleDeleteModal }) => {
+  const [deletePassword, setDeletePassword] = useState('');
+  const [deletionStatus, setDeletionStatus] = useState('Pending');
+
+  useEffect(() => {
```

Comment:
It seems like this useEffect doesn't run on successful deletion of the user profile.

Reply:
I figured out it was navigating to Landing but it wasn't closing the modal after signing the user out

Reply:
great! seems like an easy fix then!